packages/actor-query-operation-project/lib/ActorQueryOperationProject.ts

```diff
@@ -57,12 +58,15 @@ export class ActorQueryOperationProject extends ActorQueryOperationTypedMediated
     bindingsStream = bindingsStream.transform({
       map(bindings: Bindings) {
         blankNodeCounter++;
+        const scopedBlankNodesCache = new Map<string, RDF.BlankNode>();
         return <Bindings> bindings.map(term => {
-          if (term && term.termType === 'BlankNode') {
-            if (term instanceof BlankNodeScoped) {
-              return new BlankNodeScoped(`${term.value}${blankNodeCounter}`, term.skolemized);
```

Comment:
Unless I'm missing something, we also need to keep `BlankNodeScoped`, right?

Reply:
Why? A blank node scoped to a source is still a blank node that is not scoped to a single set of bindings. Modifying its label with a per-binding counter would break correlations, would it not?

Reply:
We don't need to modify the label, we just need to ensure that we keep the skolemized term. Because we don't want to break cases like this that depend on these skolemized terms: https://github.com/solid/query-ldflex/issues/34

Reply:
Oooh my bad, I had misinterpreted the question. Will do!

Reply:
Doesn't the current version of the code already preserve `BlankNodeScoped`, though? All terms are returned as-is with the exception of `BlankNodeBindingsScoped` instances.

Reply:
Ah yes, you're right, I didn't interpret the code correctly.
Then this change makes things even a bit cleaner :-)